app/commands/exercise/export_solutions_to_zip_file.rb

```diff
@@ -3,12 +3,15 @@
 class Exercise::ExportSolutionsToZipFile
   include Mandate
 
+  def self.num_submissions = 500
+
   initialize_with :exercise
 
   def call
     file_stream = Zip::OutputStream.write_buffer do |zip|
       solutions.each.with_index do |solution, idx|
-        submission = solution.submissions.first
+        submission = solution.iterations.first.submission
+        next unless submission
```

Comment:
When would this condition be true? Just asking as this would cause the output to potentially not have 500 records. Which is fine, but might be odd.

Reply:
I was thinking if the first iteration is deleted. But we should probably change it to be `solution.iterations.first&.submission` for that case.

Reply:
It probably doesn't really matter much

Reply:
I think this will definitely fail without the guard for deleted iterations as that happens all the time. I'll update.